### ISSUE TITLE
Albert accuracy is bad

### ISSUE THREAD

**Issue by sgligorijevicTT** [tenstorrent/tt-xla#350]
This issue pulls double duty to both track progress and document investigation results so they don't get lost in an internal doc somewhere.
Although the focus is on albert, underlying issues likely affect most transformers.

Albert atol on main right now is ~131000.

Debugging uncovered multiple issues:

1) Positional embedding has atol of ~0.15 and input embedding atol of ~0.2, token type embedding is good with atol <0.0003. 
2) Layernorm in the embedding layer shoots up the atol to ~16, up from atol of ~0.3 on the input
3) `jnp.ones_like` returns all zeros instead of all ones, as this is used for generating the attention mask, it makes attention generate nonsense, and attn shoots up our atol to ~131000

With a workaround for 3) the atol slowly climbs from 8 before the first transformer block to 22 after the last block, then shoots up to 34 on the logits. It's interesting that the accuracy is improved by something between the embedding LN and the first block...

Components pending investigation:
1) tanh - metal has some issues opened regarding tanh accuracy and we are seeing nondeterministic failures on allclose comparison for tanh.

**Comment by mrakitaTT** [comment]
Potentially related: https://github.com/tenstorrent/tt-xla/issues/342

**Comment by sgligorijevicTT** [comment]
Albert passed
